spec/javascripts/packs/submit-with-spinner-spec.js

```diff
@@ -0,0 +1,44 @@
+import { screen } from '@testing-library/dom';
+import { useCleanDOM } from '../support/dom';
+
+describe('submit-with-spinner', () => {
+  useCleanDOM();
+
+  async function initialize({ withForm = true } = {}) {
+    const parent = withForm
+      ? document.body.appendChild(document.createElement('form'))
+      : document.body;
+
+    parent.innerHTML = `
+      <button type='submit' class='btn btn-primary btn-wide sm-col sm-col-6'>
+        Continue
+      </button>
+      <div class='pl1 sm-col sm-col-3'>
+      <div class='spinner hidden' id='submit-spinner'>
+        <img height="50" width="50" alt="Loading spinner" src="/assets/wait.gif">
+      </div>
+      <div class='clearfix'></div>
+    `;
+
+    delete require.cache[require.resolve('../../../app/javascript/packs/submit-with-spinner')];
+    await import('../../../app/javascript/packs/submit-with-spinner');
+  }
+
+  it('gracefully handles absence of form', async () => {
+    await initialize({ withForm: false });
+  });
```

Comment:
is this just a "it doesn't blow up" kind of spec?

Reply:
Yeah, the general thinking was: We have a pretty reasonable guarantee that `#submit-spinner` is on the page, because the script is printed immediately adjacent to it. And while we largely assume that someone would only render the `submit_with_spinner` partial within a form, there's not nearly as strong a guarantee. So the test here is providing some reasonable assurance, and essentially maps to the decision to include an optional chaining operator  at...

https://github.com/18F/identity-idp/blob/65b3117116cac3c90e9af2f76c60441b5bf983a4/app/javascript/packs/submit-with-spinner.js#L5